packages/react-router/modules/Route.js

```diff
@@ -66,7 +66,7 @@ class Route extends React.Component {
     const { route } = router
     const pathname = (location || route.location).pathname
 
-    return path ? matchPath(pathname, { path, strict, exact, sensitive }) : route.match
+    return path ? matchPath(pathname, { path, strict, parent, sensitive }) : route.match
```

Comment:
Just to note, this PR maintains non-exact matching for `<Route component={Thing} />` since it will just re-use the parent `match`.